### PR TITLE
Update commitlint v17 -> v19

### PR DESCRIPTION
**Pull request by rriski**
v19 contains breaking changes, see:
- https://github.com/conventional-changelog/commitlint/blob/master/CHANGELOG.md#1900-2024-02-27

### CODE REVIEW

File: actions/commitlint/commitlint.test.ts [hunk 1]
```diff
@@ -4,8 +4,8 @@ import { TrunkActionDriver } from "tests/driver";
 
 const preCheck = (driver: TrunkActionDriver) => {
   driver.writeFile(
-    "commitlint.config.js",
-    "module.exports = {extends: ['@commitlint/config-conventional']}",
+    "commitlint.config.mjs",
+    "export default {extends: ['@commitlint/config-conventional']}",
```
Comment (rriski) [inline actions/commitlint/commitlint.test.ts:8]: Not sure how renaming the file and using `export default` should be handled by trunk when upgrading to this version. I.e. how to handle breaking changers in tools supported by trunk?

Comment (rriski) [reply actions/commitlint/commitlint.test.ts:8]: I suppose its up to the user to configure the commitlint config file. I’ll need to update the commitlint readme here: https://github.com/trunk-io/plugins/tree/main/actions/commitlint

Comment (TylerJang27) [reply actions/commitlint/commitlint.test.ts:8]: Thanks for the PR @rriski and for bringing this to our attention! I'll speak with the team about if there is any better way we can handle the update process for users. I think we will probably want to go ahead and land this, and also update the readme instructions, as you mentioned, but I'll follow-up again shortly.

**Review by TylerJang27 — APPROVED** [review]
Thanks for the PR! I will call this out in our next release as a breaking change

**Pull request merged**